app/Jobs/CreateRockTheVotePostInRogue.php

```diff
@@ -39,35 +39,42 @@ public function __construct($record)
      */
     public function handle(Rogue $rogue)
     {
-    	$shouldProcess = $this->scrubRecord($this->record);
+    	$isTestData = $this->isTestData($this->record);
 
-        if ($shouldProcess) {
+        if (!$isTestData) {
             info('progress_log: Processing: ' . $this->record['Email address']);
 
-            $referralCode = $this->record['Tracking Source'];
-            $referralCodeValues = $this->parseReferralCode($referralCode);
+            $referralCodeValues = $this->parseReferralCode($this->record['Tracking Source']);
+            info('Referral code: ' . implode(', ', $referralCodeValues));
 
-            $user = $this->getOrCreateUser($this->record, $referralCodeValues);
+            $userId = $referralCodeValues['northstar_id'];
+            $campaignId = (int) $referralCodeValues['campaign_id'];
 
-            // @TODO: If we write more functions that are identical across all voter reg imports, pull out into a ImportsVoterReg trait
-            $this->updateNorthstarStatus($user, $this->translateStatus($this->record['Status'], $this->record['Finish with State']));
+            if (!isset($userId) || empty($userId)) {
+                $userId = $this->getOrCreateUserId($this->record);
+            }
 
-            $post = $rogue->getPost([
-                'campaign_id' => (int) $referralCodeValues['campaign_id'],
-                'northstar_id' => $user->id,
-                'type' => 'voter-reg',
+            // @TODO: If we write more functions that are identical across all voter reg imports, pull out into a ImportsVoterReg trait
+            $this->updateNorthstarStatus($userId, $this->translateStatus($this->record['Status'], $this->record['Finish with State']));
+
+            $postType = 'voter-reg';
+            // Check if post exists.
+            $fetchPostRes = $rogue->getPost([
```

Comment:
Can we rename this variable to `$existingPosts`? I'm not sure it's explicitly stated anywhere, but we tend to avoid abbreviations wherever possible.

Reply:
This is unrelated to this change, but it'd also be nice to rename this method to `$rogue->getPosts`. I didn't realize this was returning an array of posts until I'd read further down this file.